src/data_access/API/GOStationApiClass.java

```diff
@@ -0,0 +1,57 @@
+package data_access.API;
+
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class GOStationApiClass implements TrainApiInterface {
+
+    private final String PARTIAL_API_URL = "OpenDataAPI/api/V1";
+    public GOStationApiClass () {
+    }
+    public List<String> retrieveStationAmenities(String stationId){
+        OkHttpClient client = new OkHttpClient().newBuilder()
+                .build();
+        HttpUrl httpUrl = new HttpUrl.Builder()
+                .scheme("http")
+                .host("api.openmetrolinx.com")
+                .addPathSegment(PARTIAL_API_URL)
+                .addPathSegment("Stop")
+                .addPathSegment("Details")
+                .addPathSegment(stationId) //getting station information for the station with the id denoted by stationId
+                .addQueryParameter("key", API_KEY)
+                .build();
+
+        Request request = new Request.Builder()
+                .url(httpUrl)
+                .addHeader("content-type", "application/json")
+                .build();
+        try {
+            // TODO: Do we need additional error checking based on error codes in Metadata?
```

Comment:
From my perspective, most cases returning error codes (i.e. user error) would already be handled by the 'station not found' popup. Some edge cases worth considering:

- Server hosting the GO API goes down, returns 503 Unavailable
- Our API key gets deactivated suddenly, returns 401 Unauthorized
- GO Transit removes/renames a station/station code, so when a user inputs a station, it will be in the data file, but API queries will result in a 204 No Content